docs/dev/framework/filesystem/config.md

```diff
@@ -0,0 +1,120 @@
+```php
+// src/ContaoManager/Plugin.php
+namespace App\ContaoManager;
+
+use Contao\CoreBundle\DependencyInjection\Filesystem\FilesystemConfiguration;
+use Contao\ManagerPlugin\Config\ConfigPluginInterface;
+use Symfony\Component\Config\Loader\LoaderInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+
+class Plugin implements ConfigPluginInterface
+{
+    public function registerContainerConfiguration(LoaderInterface $loader, array $managerConfig)
+    {
+        $loader->load(static function (ContainerBuilder $container) {
+            $config = new FilesystemConfiguration($container);
+            
+            // Configure the filesystem here
+        });
+    }
+}
+```
+
+## Reference
+|  |  |
+|-|-|
+| `addVirtualFilesystem` | Add a new virtual filesystem. Setting the `$name` to 'foo' will create a `contao.filesystem.virtual.foo` service and additionally enable constructor injection with an argument `VirtualFilesystemInterface $fooStorage` if autowiring is available. The `$prefix` defines the directory the instance should be scoped to. Optionally setting `$readonly` to true, will prevent mutations being made to resources and metadata. |
+| `mountAdapter` | Creates a new adapter and mounts it under the given `$mountPath` in the `MountManager`. The `$adapter` and `$options` can be set analogous to the configuration of the [Flysystem Symfony bundle][FlysystemBundle]. Alternatively you can pass in an id of an already existing filesystem adapter service. If you do not set a `$name`, the id/alias for the adapter service will be derived from the mount path. |
+| `mountLocalAdapter` | Use this shortcut method if you want to create a local adapter. Internally this is using `mountAdapter` but already resolves project root relative paths and parameter placeholders for you. |
+| `registerDbafs` | Registers a custom `$dbafs` service definition in the `DbafsManager` under a given `$pathPrefix`. |
+| `addDefaultDbafs` | Creates a default DBAFS implementation as a service and internally calls `registerDbafs` on it. We recommend to use this method if the default implementation fits your needs. Set the `$tableName` and `$hashFunction` to be used and optionally disable tracking last modified states by setting `$useLastModified` to false. |
+
+When using `addVirtualFilesystem` and `addDefaultDbafs()`, the preconfigured `Definition` is returned. If you want to
+tweak it (e.g. if you want to adjust the DBAFS defaults), just add methods calls yourself:
+
+```php
+// Manually adjust a configuration, if needed
+$config
+  ->addDefaultDbafs(…)
+  ->registerMethodCall('setMaxFileSize', [1048576])
+;
+
+```
+
+### Example
+Here is how you could make your automatic database backups be stored on a remote FTP server:
+```php
+// Mount an FTP adapter to `/backups`.
+$config->mountAdapter(
+    'sftp',         
+    ['host' => 'example.com', 'port' => 22, 'username' => 'foobar', 'password' => 's3cr3t'],
+    'backups'
+);
+```
+
```

Comment:
```suggestion

The mount options in the second argument are directly passed to the Flysystem adapter. See <https://github.com/thephpleague/flysystem-bundle/blob/master/docs/3-interacting-with-ftp-and-sftp-servers.md#sftp> for more information.

```

Reply:
Done in d8a764f86f6bf019041982d26c95d62c1aea890b